darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+from darshan.report import *
+
+import matplotlib
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable 
+import numpy as np
+
+def plot_dxt_heatmap(report, 
+        xbins=10, ybins=None, 
+        group_by="rank", 
+        mods=None, ops=None, 
+        display_values=False, cmap=None, figsize=None, 
+        amplify=False):
+    """
+    Generates a heatmap plot from a report with DXT traces.
```

Comment:
@nawtrey How does your DXT heatmap compare with the result below using this feature branch? The test case, available in the repo test suite, is so simple that it should provide a fairly obvious comparison point.

```python
import darshan
import darshan.backend.cffi_backend as backend
from darshan.experimental.plots.plot_dxt_heatmap import plot_dxt_heatmap

# use the super-simple DXT trace currently in the pydarshan
# test suite as a first pass on DXT heat map plot appearance
logfile = '/home/tyler/github_projects/darshan/darshan-util/pydarshan/tests/input/sample-dxt-simple.darshan'
report = darshan.DarshanReport(logfile, read_all=True)

fig = plot_dxt_heatmap(report=report,
                 mods=['DXT_POSIX'])

fig.savefig('test_jakob_dxt_heatmap.png', dpi=300)
```

![test_jakob_dxt_heatmap](https://user-images.githubusercontent.com/7903078/120546081-ac73e600-c3ac-11eb-8215-c0761fbe855b.png)

Reply:
After you do that comparison, here's a more complex DXT-enabled log file (
[dxt-log2](https://github.com/darshan-hpc/darshan/files/6587813/treddy_mpi-io-test_id4373053_6-2-60198-9815401321915095332_1.darshan.gz)
; it has been gzipped, so unzip it first) you can use to compare with Jakob, this time with the plot output below:

![test_jakob_dxt_heatmap](https://user-images.githubusercontent.com/7903078/120561732-54e07500-c3c2-11eb-8e2f-5d83db19d0bd.png)

Reply:
And 
[an even larger/more complex gzipped/DXT-enabled log file ](https://github.com/darshan-hpc/darshan/files/6587863/treddy_mpi-io-test_id4373096_6-2-61446-92088258725910398_2.darshan.gz) that produced the plot below:

![test_jakob_dxt_heatmap](https://user-images.githubusercontent.com/7903078/120563345-9161a000-c3c5-11eb-8dfc-092b3414be3e.png)

Reply:
Pushing up to 640 MPI ranks with this [gzipped log file](https://github.com/darshan-hpc/darshan/files/6588042/treddy_mpi-io-test_id17541138_6-2-65169-14537227514913022501_15.darshan.gz)
should be useful to compare the dynamic sizing capabilities of the two approaches:

![test_jakob_dxt_heatmap](https://user-images.githubusercontent.com/7903078/120567467-17ceaf80-c3cf-11eb-98f0-7ca63d52a7e6.png)

Reply:
And with `2240` MPI ranks on an `aarch64` cluster with this [gzipped log file](https://github.com/darshan-hpc/darshan/files/6593805/treddy_mpi-io-test_id545458_6-3-43062-1247823192538596076_13.darshan.gz):

![test_jakob_dxt_heatmap](https://user-images.githubusercontent.com/7903078/120698962-fde3aa00-c46c-11eb-9abd-e8b6421565f2.png)

Reply:
One thing that might be more intuitive is to use the actual time in seconds for the `x` tick labels, instead of the physical bin numbers.

For example,

```diff
@@ -119,6 +131,11 @@ def plot_dxt_heatmap(report,
     ax.set_title(f"DXT Heatmap mods={mods}, ops={ops}")
     ax.set_ylabel(ylabel)
     ax.set_xlabel(f"Time (binsize={runtime/xbins} seconds)")
+    binsize = runtime/xbins
+    xticks = map(str, ax.get_xticks() * binsize)
+    ax.set_xticklabels(xticks)
```

![image](https://user-images.githubusercontent.com/7903078/120716431-41e1a980-c483-11eb-858c-f778e47cb15e.png)

Reply:
Yeah, I think it's helpful to retain the exact application timing, rather than in terms of bins, so the duration of I/O activity is more clear (i.e., "there was heavy I/O activity for 15 seconds" vs "there was heavy I/O activity for 4 bins and each bin is 3.5 seconds").

Reply:
I think the discussion of these has since been continued in https://github.com/darshan-hpc/darshan/pull/396